### DISCUSSION TITLE
Duplicate Devices Redux

### DISCUSSION THREAD

**Discussion by Lorccan** [homebridge-simplisafe3/homebridge-simplisafe3#412] [Q&A]
I asked a question recently: https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/410 about unresponsive SimpliSafe duplicates appearing in HomeKit.

Although the original answer (deleting cached accessories) fixed the problem, they have reappeared. Given that when I removed and reinstalled the plugin, the authorisation was still present I'm wondering whether this could have any bearing on the problem.

Any advice on how to proceed would be most welcome!

**Comment by shamoon** [comment]
Hmm, nothing obvious I can think of and this isn't something ive seen others having. Perhaps there's some sort of issue with your HB installation. To answer your question though, the auth being preserved shouldn't really be relevant or a problem.

Also, note that removing a plugin doesnt necessarily remove accessories (if I recall youre given the option to)

Accessories are compared based on the serial number, which is a stable value provided from the SS API...

**Comment by Lorccan** [comment]
Thanks for responding again. I removed the Simplisafe plugin from HomeKit (which removed the _real_ accessories), removed a single "phantom" accessory which caused the rest of them to disappear as well (NOTE: these were not associated with any bridge/hub, so were shown as standalone accessories), restarted homebridge, deleted cached accessories again (which also restarted homebridge) and then re-added the plugin to HomeKIt. (At this point, I haven't actually removed the plugin from homebridge first as I really don't think that's where the problem originated.)

I'm going to leave this open for a couple of days as all looks correct now, but it did last time too!  I'll come back with, I hope, some good news!

**Comment by Lorccan** [comment]
Unfortunately, I have precisely the same outcome as previously: two duplicates of every SimpliSafe device (excepting one which only has a single duplicate) have appeared in the Default Room each with "No Response". I've had a look at the serial numbers: in each case the duplicates match those of the 'real' responsive accessory. What's even more strange, though, is that when I look (in Home app) at the bridge for a contact sensor, it takes me to a (SimpliSafe) smoke detector. When I then look at the details for the 2 duplicates of this smoke detector, one is a standalone device (with no bridge) and the other is a bridge to the first one!

My homebridge installation has been running for years without problems and this is the only plugin that's giving any issues at all.

I suppose I can put up with the unresponsive accessories, but it seems strange that I am the only one getting this problem. I don't know enough about SimpliSafe or how the plugin works with it to know whether deleting all the accessories from it and starting over would help, but I suspect it would not.

**Reply by shamoon** [reply]
Yea, it's very weird, sorry it's giving you trouble. I wouldn't bother removing devices in SS (that sounds awful).

I actually think this might be an iCloud thing not a Homebridge thing. Have you googled the issue in general ("homebridge duplicate accessories / devices")? There are like majillions of results, so not sure which would be most helpful. I think I've heard of signing out of  iCloud hubs (maybe devices)?

other than that is really the nuclear options. Remove the home (in HomeKit), remove the entire homebridge install, etc. but of course those are all a PITA. I can recommend that https://controllerforhomekit.com takes some of the pain out of restoring...

Also, maybe:

- What does your config look like?
- Do the devices just eventually pop up or is it only after e.g. a HB restart?
